test/Npgsql.Tests/ConnectionTests.cs

```diff
@@ -528,6 +528,76 @@ public async Task TimezoneConnectionParam()
 
         #endregion Timezone
 
+        #region ConnectionString - Host
+
+        [TestCase("127.0.0.1", ExpectedResult = new []{"tcp://127.0.0.1:5432"})]
+        [TestCase("127.0.0.1:5432", ExpectedResult = new []{"tcp://127.0.0.1:5432"})]
+        [TestCase("::1", ExpectedResult = new []{"tcp://::1:5432"})]
+        [TestCase("[::1]", ExpectedResult = new []{"tcp://[::1]:5432"})]
+        [TestCase("[::1]:5432", ExpectedResult = new []{"tcp://[::1]:5432"})]
+        [TestCase("localhost", ExpectedResult = new []{"tcp://localhost:5432"})]
+        [TestCase("localhost:5432", ExpectedResult = new []{"tcp://localhost:5432"})]
+        [TestCase("127.0.0.1,127.0.0.1:5432,::1,[::1],[::1]:5432,localhost,localhost:5432",
+            ExpectedResult = new []
+            {
+                "tcp://127.0.0.1:5432",
+                "tcp://127.0.0.1:5432",
+                "tcp://::1:5432",
+                "tcp://[::1]:5432",
+                "tcp://[::1]:5432",
+                "tcp://localhost:5432",
+                "tcp://localhost:5432"
+            })]
+        [Test, IssueLink("https://github.com/npgsql/npgsql/issues/3802"), NonParallelizable]
+        public async Task<string[]> ConnectionString_Host(string host)
+        {
+            var numberOfHosts = host.Split(',').Length;
+            if (numberOfHosts > 1)
+            {
+                if (IsMultiplexing)
+                    throw new SuccessException("Multiple hosts in connection string is ignored for Multiplexing");
+                // We reset the cluster's state for multiple hosts
+                // Because other tests might have marked some of the hosts as disabled
+                ClusterStateCache.Clear();
```

Comment:
Any thoughts on test parallelization here? There's a race condition where a host may get marked as disabled after this is called - at the very least maybe put a [NonParallelizable] on this test.

Reply:
Sorry, missed it.